backend/pkg/portforward/handler_test.go

```
@@ -51,21 +52,46 @@ func getDefaultKubeConfigPath(t *testing.T) string {
 	return filepath.Join(homeDirectory, ".kube", "config")
 }
 
-//nolint:funlen
+// findPodWithPort finds a suitable pod to port-forward to, returning its name and port.
+func findPodWithPort(t *testing.T, clientSet interface{}) (podName, targetPort string) {
+	podList, err := clientSet.(kubernetes.Interface).CoreV1().
+		Pods("headlamp").List(context.Background(), metav1.ListOptions{})
+	require.NoError(t, err)
+	require.NotEmpty(t, podList.Items)
+
+	// Select the pod and make sure it has a port to forward to
+	for _, pod := range podList.Items {
+		if len(pod.Spec.Containers) > 0 && len(pod.Spec.Containers[0].Ports) > 0 {
+			return pod.Name, fmt.Sprint(pod.Spec.Containers[0].Ports[0].ContainerPort)
+		}
+	}
+
+	return "", ""
```

Returning empty strings on failure ? This hides the errors and makes debugging hard. Just return an error if the function can fail.

---

Thank you for reviewing my PR. I guess the tests use empty strings, and earlier implementations also had the same.